### PR TITLE
feat: Reduce blast radius of errors in MarketMap

### PR DESCRIPTION
**Pull request by Eric-Warehime**
Adds a method to the marketmap type, `GetValidSubset`. This is used in the update lifecycle in order to be able to accept updates to the marketmap which do not fully pass ValidateBasic.

It will attempt to remove any `ProviderConfig` with an invalid `NormalizeByPair`, and then remove any `Market` which fails to pass `ValidateBasic`. 

If a market has some valid ProviderConfigs it can still run as long as it still satisfies the min provider count.

### CODE REVIEW

File: x/marketmap/types/market.go [hunk 1]
```diff
@@ -40,6 +40,48 @@ func (mm *MarketMap) ValidateBasic() error {
 	return nil
 }
 
+// GetValidSubset outputs a MarketMap which contains the maximal valid subset of this MarketMap.
+//
+//	In particular, this will eliminate anything which would otherwise cause a failure in ValidateBasic.
+//	The resulting MarketMap should be able to pass ValidateBasic.
+func (mm *MarketMap) GetValidSubset() MarketMap {
```
Comment (aljo242) [inline x/marketmap/types/market.go]: I feel like this function should return an `error` as well and call `Validate` to ensure that the returned map is valid

**Review by aljo242 — APPROVED** [review]
otherwise lgtm

File: x/marketmap/types/market.go (hunk 1 above)
Comment (aljo242) [reply x/marketmap/types/market.go]: There is a case where there is no valid subset and we should just return `nil, error`

Comment (Eric-Warehime) [reply x/marketmap/types/market.go]: I think you're right--if the entire set of markets is invalid we should just keep everything the same. Returning an error will do that while I think returning an empty subset would just stop fetching any prices at all.

**Review by technicallyty — APPROVED** [review]
LGTM pending @aljo242's suggestion

**Pull request merged**